python/lsst/pipe/base/config.py

```diff
@@ -284,6 +295,15 @@ class ResourceConfig(pexConfig.Config):
                                   doc="Minimal number of cores needed by task.")
 
 
+def _templateNameStorageDecorator(func):
+    storedParamsDict = {}
```

Comment:
I think this variable ends up effectively being a class attribute, by virtue of being a local variable in a closure defined at class scope.  I think what you want is a new instance attribute, which I *think* you can do by just dropping the decorator and doing something like the following in `formatTemplateNames` directly:
```
if not hasattr(self, "_storedParamsDict"):
    self._storedParamsDict = {}
self._storedParamsDict.update(templateParamsDict)
templateParamsDict.update(self._storedParamsDict)
```

A completely different (and possibly cleaner approach) would be to use [Formatter.parse](https://docs.python.org/3.6/library/string.html#string.Formatter.parse) to skip substitution for fields that don't actually have the parameters given in a particular call.